client/apiversion.go

```diff
@@ -151,24 +169,46 @@ func (c *authenticatingClient) getAPIVersions(serviceCatalogURL string) (*apiURL
 	// API version-specific base URL.
 	var pathParts, origPathParts []string
 	if url.Path != "/" {
-		// The rackspace object-store endpoint triggers the version removal here,
-		// so keep the original parts for object-store and container endpoints.
+		// If a version is included in the serviceCatalogURL, the
+		// part before the version will end up in url, the part after
+		// the version will end up in pathParts.  origPathParts is a
+		// special case for "object-store"
 		// e.g. https://storage101.dfw1.clouddrive.com/v1/MossoCloudFS_1019383
+		// https://x.x.x.x/image
+		// https://x.x.x.x/cloudformation/v1
+		// https://x.x.x.x/compute/v2/9032a0051293421eb20b64da69d46252
+		// http://y.y.y.y:9292
+		// http://y.y.y.y:8774/v2/010ab46135ba414882641f663ec917b6
 		origPathParts = strings.Split(strings.Trim(url.Path, "/"), "/")
 		pathParts = origPathParts
-		if _, err := parseVersion(pathParts[0]); err == nil {
-			pathParts = pathParts[1:]
+		found := false
+		for i, p := range pathParts {
+			if _, err := parseVersion(p); err == nil {
+				found = true
+				if i == 0 {
```

Comment:
Debug statements would be useful here for each condition - if we have to do this again, it would be nice to understand how juju parsed it.

Reply:
Can you also provide an example URL or format for each case?

Reply:
added logger.tracef and additional comments